storage/pebble/blocks.go

```diff
@@ -103,36 +98,87 @@ func (b *Blocks) GetByID(ID common.Hash) (*types.Block, error) {
 
 	height, err := b.store.get(blockIDToHeightKey, ID.Bytes())
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get block by ID: %w", err)
+	}
+
+	blk, err := b.getBlock(blockHeightKey, height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get block by height: %w", err)
+	}
+
+	return blk, nil
+}
+
+func (b *Blocks) LatestEVMHeight() (uint64, error) {
+	return b.getHeight(latestEVMHeightKey)
+}
+
+func (b *Blocks) FirstEVMHeight() (uint64, error) {
+	return b.getHeight(firstEVMHeightKey)
+}
+
+func (b *Blocks) LatestCadenceHeight() (uint64, error) {
+	b.mux.RLock()
+	defer b.mux.RUnlock()
+
+	val, err := b.store.get(latestCadenceHeightKey)
+	if err != nil {
+		if errors.Is(err, errs.NotFound) {
+			return 0, errs.NotInitialized
+		}
+		return 0, fmt.Errorf("failed to get latest cadence height: %w", err)
 	}
 
-	return b.getBlock(blockHeightKey, height)
+	return binary.BigEndian.Uint64(val), nil
 }
 
-func (b *Blocks) LatestHeight() (uint64, error) {
-	return b.getHeight(latestHeightKey)
+// InitCadenceHeight sets the latest Cadence height
+func (b *Blocks) InitCadenceHeight(height uint64) error {
```

Comment:
Should we allow overriding the initial cadence height? I am not sure if it has any value.

Reply:
I don't think we should as a safety measure in case the node restarts and it keeps the same flags it would overwrite the db. Generally overwriting I feel should be manual and explicit. But I do feel the whole bootstrap process should be further improved.